File: rest/src/main/java/de/philgenstock/choremaster/household/HouseHoldEntity.java
package de.philgenstock.choremaster.household;

import de.philgenstock.choremaster.persistence.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "household")
public class HouseHoldEntity extends BaseEntity {

  private String name;
}

File: choremaster-rest/rest/src/main/kotlin/de/philgenstock/choremaster/household/HouseHoldEntity.kt
package de.philgenstock.choremaster.household

import de.philgenstock.choremaster.user.UserEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "household")
data class HouseHoldEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    val name: String,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    val owner: UserEntity,
)
